lib/monitor/monitor_test.go

```diff
@@ -3,91 +3,64 @@ package monitor
 import (
 	"bytes"
 	"fmt"
+	"github.com/dedis/cothority/lib/dbg"
 	"strconv"
-	"strings"
 	"testing"
 	"time"
-
-	"github.com/dedis/cothority/lib/dbg"
 )
 
-func TestMonitor(t *testing.T) {
-	defer dbg.AfterTest(t)
-
+// setupMonitor launches a basic monitor with a created Stats object
+// When finished with the monitor, just call `End()`
+func setupMonitor(t *testing.T) (*Monitor, *Stats) {
 	dbg.TestOutput(testing.Verbose(), 2)
 	m := make(map[string]string)
 	m["servers"] = "1"
-	m["hosts"] = "1"
 	stat := NewStats(m)
-	fresh := stat.String()
 	// First set up monitor listening
 	mon := NewMonitor(stat)
 	go mon.Listen()
 	time.Sleep(100 * time.Millisecond)
 
 	// Then measure
-	err := ConnectSink("localhost:" + strconv.Itoa(DefaultSinkPort))
+	err := ConnectSink("localhost:" + strconv.Itoa(mon.SinkPort))
+	EnableMeasure(true)
 	if err != nil {
-		t.Error(fmt.Sprintf("Error starting monitor: %s", err))
-		return
-	}
-
-	meas := NewMeasure("round")
-	meas.Measure()
-	time.Sleep(200 * time.Millisecond)
-	meas.Measure()
-	EndAndCleanup()
-	time.Sleep(100 * time.Millisecond)
-	updated := stat.String()
-	if updated == fresh {
-		t.Error("Stats not updated ?")
+		t.Fatal(fmt.Sprintf("Error starting monitor: %s", err))
 	}
+	return mon, stat
 }
 
 func TestReadyNormal(t *testing.T) {
-	defer dbg.AfterTest(t)
+	//defer dbg.AfterTest(t)
```

Comment:
mhh right. I should put it back. it was during the testing phase something was wrong and I was getting annoyed by the output :D